app/lib/AuthContext.tsx

```diff
@@ -0,0 +1,51 @@
+'use client';
+
+import {
+  createContext,
+  useContext,
+  useState,
+  ReactNode,
+  useEffect,
+} from 'react';
+
+interface AuthContextType {
+  isAuthenticated: boolean;
+  login: () => void;
+  logout: () => void;
+}
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
```

Comment:
In most cases if you want to set something to like undefined, you would better use null. 
It is like a rule of thumb. If you aim to put something that does not exists use null. 
Undefined is something that code throw to you